features/employers/benefit_applications/admin_reinstate_benefit_application_steps.rb

```diff
@@ -8,18 +8,74 @@
   disable_feature :benefit_application_reinstate
 end
 
-And(/^initial employer ABC Widgets has updated (.*) effective period for (.*)$/) do |aasm_state|
-  if aasm_state == 'terminated'
-    employer_profile.benefit_applications.first.workflow_state_transitions << WorkflowStateTransition.new(from_state: 'active', to_state: 'terminated', event: 'terminate!')
-    term_ba = employer_profile.benefit_applications.first
-    start_on = term_ba.benefit_sponsor_catalog.effective_period.min
-    end_on = term_ba.benefit_sponsor_catalog.effective_period.max
+And(/^initial employer ABC Widgets has updated (.*) effective period for reinstate$/) do |aasm_state|
+  if aasm_state == 'retroactive_canceled'
```

Comment:
- No need to verify for terminated block because before step is already creating a terminated benefit application with required effective_period dates. retroactive_canceled is newly introduced aasm_state so it required that effective_period to be updated
 - The original intent was same as new change. I saw this step was failing so I had to fix it and this makes the step more precise and easier to understand. 
 Originally: 
 ```
 initial employer ABC Widgets has updated (.*) effective period for (.*) 
 for steps:
 initial employer ABC Widgets has updated (.*) effective period for reinstate
   initial employer ABC Widgets has updated (.*) effective period for revise end date
 ```

Reply:
i) "No need to verify for terminated" -- if you no longer needed terminated block, do you still need `aasm_state == 'retroactive_canceled'` check ?
ii) i'm a bit confused here - 

your step def says - `initial employer ABC Widgets has updated (.*) effective period for reinstate`  and you're calling this step for `revise`.
Instead you can have a step def saying ` `initial employer ABC Widgets has updated (.*) effective period` something like this and then use it for both `reinstate` and `revise`

Reply:
1. 
```
initial employer ABC Widgets has updated (.*) effective period for reinstate do |aasm_state| 
```
This above step is used for different examples. Regular expression (.*) indicates words like terminated/termination_pending/retroactive_canceled so I want to update only when the regular expression word is retroactive_canceled. 
2. Yes, I'm calling`initial employer ABC Widgets has updated (.*) effective period for reinstate` because to avoid redundancy. Even revise uses the same step definition.

Reply:
yes, you're understanding on `(.*)` is right. But if you hard code it only for `retroactive_canceled ` what's the purpose here?
And, why do we need to call this for other states, if this doesn't do anything for them ?

>  Yes, I'm calling initial employer ABC Widgets has updated (.*) effective period for reinstate because to avoid redundancy. Even revise uses the same step definition.

-- right, you still have redundant code here when you're calling one step inside another step.
you can name this step generic without using reinstate/revise in it so you don't need an additional step definition to make this call, you can just re-use it. This simplifies it even further.